packages/aws-library/src/aws_library/s3/client.py

```diff
@@ -0,0 +1,72 @@
+import contextlib
+import logging
+from dataclasses import dataclass
+from typing import cast
+
+import aioboto3
+import botocore.exceptions
+from aiobotocore.session import ClientCreatorContext
+from botocore.client import Config
+from models_library.api_schemas_storage import S3BucketName
+from pydantic import AnyUrl, parse_obj_as
+from settings_library.s3 import S3Settings
+from types_aiobotocore_s3 import S3Client
+
+from .errors import S3RuntimeError
+
+_logger = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class SimcoreS3API:
+    client: S3Client
+    session: aioboto3.Session
+    exit_stack: contextlib.AsyncExitStack
+
+    @classmethod
+    async def create(cls, settings: S3Settings) -> "SimcoreS3API":
+        session = aioboto3.Session()
+        session_client = session.client(
+            "s3",
+            endpoint_url=settings.S3_ENDPOINT,
+            aws_access_key_id=settings.S3_ACCESS_KEY,
+            aws_secret_access_key=settings.S3_SECRET_KEY,
+            aws_session_token=settings.S3_ACCESS_TOKEN,
+            region_name=settings.S3_REGION,
+            config=Config(signature_version="s3v4"),
+        )
+        assert isinstance(session_client, ClientCreatorContext)  # nosec
+        exit_stack = contextlib.AsyncExitStack()
+        s3_client = cast(
+            S3Settings, await exit_stack.enter_async_context(session_client)
+        )
+        return cls(s3_client, session, exit_stack)
+
+    async def close(self) -> None:
+        await self.exit_stack.aclose()
+
+    async def ping(self) -> bool:
```

Comment:
MINOR: we need to start standardizing these function calls. As we have discussed, there are different levels of checks on services. Stritcly speaking `ping` refers t a basic nextwork check to determine that the server is reachable. What you are doing here is "deeper" (i.e. you are actually "speaking to S3"). 
I would like to standardize these checks. I propose
1. ping: server or device is reachable by sendinc ICMP request and waiting for echo replies. Determines whether the server is online at the network level
2. tcp port check: testing that the server is running and listening to a given port
3. http check: involves sending a htp requires to a lightweight checkpoint and examining that it responds 200
5. service health check: sending requests to designated endpoints or APIs that provide information about the health and status of the service

Reply:
This call is basicaly there to check if the connection with S3 is established. There is the same happening in the EC2 client.
Maybe we should rename these functions to something like ```connected```

Now concerning the call I think you should use ```head_bucket``` function that is much more light weight (see services/storage/src/simcore_service_storage/s3_client.py) - that is also why I initially asked to not duplicate but move the code as these functions already exist somewhere else.

Reply:
Suggestion to at least start moving the s3 client away from storage:
1. make the S3 client in storage inherit this one
2. replace this ping function with the one in ```check_bucket_connection```
3. remove ```check_bucket_connection``` from the S3 client in storage
4. make ```create_single_presigned_download_link``` in storage use the one you created here below

Also please define at least a type alias for ```S3ObjectKey: TypeAlias = str``` especially since I see they do have restrictions (see https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-keys.html). Ideally create a Regex for that in packages/models-library/src/models_library/basic_regex.py and use it in ```class S3ObjectKey(ConstrainedStr): regex=S3_OBJECT_KEY_RE```

Also please move down the s3_exception_handler from services/storage/src/simcore_service_storage/s3_utils.py so we do not start again with all these exception handlings. I think you did a good part of the path, let's do it completely.

Sorry for being a pain here but I spent quite some time refactoring storage last time and I would not like to start over. Thanks a lot

Reply:
Already started here some improvements and consolidation with storage s3 client, will continue in separate PR